### PR TITLE
Do not direct access to array elements, iterate over them

### PR DESCRIPTION
**Pull request by rochoa**
cc @fdansv @dgaubert

### CODE REVIEW

File: lib/windshaft/utils/cartocss_utils.js [hunk 1]
```diff
@@ -51,11 +51,14 @@ function appendRulesFields(definitions, allSelectors) {
         .reduce(function(allRules, rules) {
```
Comment (dgaubert) [inline lib/windshaft/utils/cartocss_utils.js]: It's equivalent a `map`, isn't it?

Comment (dgaubert) [inline lib/windshaft/utils/cartocss_utils.js]: I mean, something like that:
```
.map(function (rules) {
  return rules;
})
```

Comment (dgaubert) [inline lib/windshaft/utils/cartocss_utils.js]: Ok. I see, we need to build a flat array.

Comment (rochoa) [inline lib/windshaft/utils/cartocss_utils.js]: Yep, that's it. We need an array with all rules. With a map we would have an array with arrays.

**Pull request merged**